### PR TITLE
[fix](tableFuncs) fix explode_json_array_funcs

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
1. explode_json_array_json accept jsonb type and return jsonb type or string type and return string type
2. fix explode_json_array return empty set issue
Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonArrayJson.java [hunk 1]
```diff
@@ -35,7 +36,8 @@
  */
 public class ExplodeJsonArrayJson extends TableGeneratingFunction implements UnaryExpression, PropagateNullable {
     public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
-            FunctionSignature.ret(JsonType.INSTANCE).args(JsonType.INSTANCE)
+            FunctionSignature.ret(JsonType.INSTANCE).args(JsonType.INSTANCE),
+            FunctionSignature.ret(VarcharType.SYSTEM_DEFAULT).args(JsonType.INSTANCE)
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonArrayJson.java:40]: signature should not have same arg types but different return type

**Review by morrySnow — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonArrayJson.java (hunk 1 above)
Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/generator/ExplodeJsonArrayJson.java]: done!

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by amorynan** [comment]
run cloud_p1

**Comment by amorynan** [comment]
run cloud_p1

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**